### PR TITLE
feat(NA): Add HTTP transport support

### PR DESCRIPTION
**Pull request by gabriel-pinto**

### CODE REVIEW

File: src/main/java/com/statful/client/StatfulMetricsOptions.java [hunk 1]
```diff
@@ -826,4 +839,21 @@ public StatfulMetricsOptions setMaxBufferSize(final int maxBufferSize) {
         this.maxBufferSize = maxBufferSize;
         return this;
     }
+
+    /**
+     * @return gets the Http path to store metrics
+     */
+    public String getMetricsPath() {
```
Comment (reistiago) [inline src/main/java/com/statful/client/StatfulMetricsOptions.java]: Thinking if the method name should include http somewhere. Thoughts?

File: src/main/java/com/statful/sender/HttpSender.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        request.end(toSendMetrics);
+    }
+
+    @Override
+    public void close(final Handler<AsyncResult<Void>> handler) {
+        final Optional<Handler<AsyncResult<Void>>> closeHandler = Optional.ofNullable(handler);
+
+        try {
+            this.client.close();
+            closeHandler.ifPresent(h -> h.handle(Future.succeededFuture()));
```
Comment (reistiago) [inline src/main/java/com/statful/sender/HttpSender.java]: h?

File: src/main/java/com/statful/sender/MetricsHolder.java [hunk 3]
```diff
@@ -91,4 +92,12 @@ private void flush(final int flushSize) {
             this.send(toBeSent);
         }
     }
+
+    Optional<String> bundleMetrics(final List<DataPoint> metrics) {
```
Comment (reistiago) [inline src/main/java/com/statful/sender/MetricsHolder.java]: this is missing javadoc no?

File: src/test/java/com/statful/client/StatfulHttpClientIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.nonNull;
+
+@RunWith(VertxUnitRunner.class)
+public class StatfulHttpClientIntegrationTest extends IntegrationTestCase {
```
Comment (reistiago) [inline src/test/java/com/statful/client/StatfulHttpClientIntegrationTest.java:25]: Are these tests "copied" from the UDP integration test? Thinking if we could share the tests somehow and just "change" the "sender".

**Review by reistiago — COMMENTED** [review]
looks good, just some small thingies

File: src/main/java/com/statful/client/StatfulMetricsOptions.java (hunk 1 above)
Comment (gabriel-pinto) [reply src/main/java/com/statful/client/StatfulMetricsOptions.java]: Makes sense, will change it

File: src/main/java/com/statful/sender/HttpSender.java (hunk 2 above)
Comment (gabriel-pinto) [reply src/main/java/com/statful/sender/HttpSender.java]: will change to handler

File: src/test/java/com/statful/client/StatfulHttpClientIntegrationTest.java (hunk 4 above)
Comment (gabriel-pinto) [reply src/test/java/com/statful/client/StatfulHttpClientIntegrationTest.java:25]: They are, the problem with that is the receiver has different logic, or do you mean having them on the same class?

Comment (reistiago) [reply src/test/java/com/statful/client/StatfulHttpClientIntegrationTest.java:25]: I'm not sure how to do it, but it looks strange to have the test case copy pasted to 2 classes :/

**Pull request merged**